client-app/src/BaseAppModel.ts

```diff
@@ -0,0 +1,31 @@
+import {HoistAppModel, XH} from '@xh/hoist/core';
+import {OauthService} from './core/svc/OauthService';
+import {MINUTES} from '@xh/hoist/utils/datetime';
+
+export class BaseAppModel extends HoistAppModel {
+    static override async preAuthAsync() {
+        if (!(await BaseAppModel.isSessionActiveOnDev())) {
+            await XH.installServicesAsync(OauthService);
+        }
+    }
+
+    override async logoutAsync() {
+        await XH.oauthService.logoutAsync();
+    }
+
+    override get supportsVersionBar(): boolean {
+        return window.self === window.top;
+    }
+
+    static async isSessionActiveOnDev(): Promise<boolean> {
```

Comment:
This name doesn't feel quite right to me - would something like `shouldSkipOauth` be a bit more expressive? I'd also add a comment to explain why we have this check, as it's a subtle issue and I don't think I would remember 😄 .

I see that the bit about the delay was copied here from the AppContainerModel implementation. That would not apply to Toolbox - believe it's for alternate SSO-ish flows where the user might be presented with a browser credentials prompt due to a filter-based auth negotiation.